### DISCUSSION TITLE
🏢 Mode expert > Enrichir la partie RIDET des entreprises

### DISCUSSION THREAD

**Discussion by adriens** [lschaeffer313/domaine-nc-mobile#109] [Polls]
# :grey_question: Contexte

Actuellement l'API des noms de domaine renvoie l'information `rid7` : 

![image](https://user-images.githubusercontent.com/5235127/210279614-18a8de83-9958-4d02-bde6-26d6614f4cfd.png)

... laquelle permet, via l'API Open Data du gouv [Entreprises actives au Ridet](https://nouvelle-caledonie.opendatasoft.com/explore/dataset/entreprises-actives-au-ridet/information), de récupérer d'autres informations très intéressantes comme : 

- code APE
- formule juridique
- libellé NAF
- code commune
- la province
- si salariés
- ...

:point_right: L'idée est d'enrichir les données du dmaine sur l'app avec ces éléments afin de permettre une expérience plus poussée.

Par exemple pour skazu, on a `rid7 = 0732206`... et donc via API on a ceci : 

![image](https://user-images.githubusercontent.com/5235127/210279920-44259d15-6361-48ef-ba7a-8dbd475e5321.png)

# :bookmark: Ressources

- [Wrapper REST simpliifié](https://dev.to/adriens/open-data-ridets-en-nouvelle-caledonie-52pl) 

# :warning: NB

Actuellement, la partie RIDET est buggée sur le site officiel : 

![image](https://user-images.githubusercontent.com/5235127/210291977-53ec2e01-f040-408f-a582-b5e481845770.png)

**Comment by adriens** [comment]
D'alleurs @lschaeffer313 , en tant qu'intéragteur d'app, tu préfères utiliser une API genre [Wrapper REST simpliifié](https://dev.to/adriens/open-data-ridets-en-nouvelle-caledonie-52pl) ou celle de Open Data :grey_question: :pray: Cela peut avoir un impact chez nous.

**Reply by lschaeffer313** [reply]
Je préférerai une API REST simple, cela est toujours plus simple à utiliser.
Avec une bonne doc si possible 😋 et un thé avec 😂

**Reply by adriens** [reply]
ta réponse me va très bien, c'est exactement le genre de feedback sur j'attendais : t'as comparé la mienne et celle d'open data ?... genre tu saurais chiffrer à la louche la compelxité de lintégration Flutter dans chaque cas ?

**Reply by lschaeffer313** [reply]
Faut que je regarde, je te fais un retour ce soir.

**Reply by adriens** [reply]
y'a pas d'urgence, de toute façon à priori on va freeze l'app mbile; C'est surtout pour savoir si de mon côté on délivre un proxy bien clean

**Reply by adriens** [reply]
hello @lschaeffer313  ping pour un premier feedback très grosse maille : inutile de faire une analyse très poussée : tu peux même me répondre avec un emoji en face de chaque approche :zap:

**Comment by adriens** [comment]
D'abord on développera une API clean.
:pause_button:
